### ISSUE TITLE
Research JHipster for the aggregator component

### ISSUE THREAD

**Issue by upkarlidder** [Call-for-Code-for-Racial-Justice/Open-Sentencing-Aggregator#24]

**Comment by amfred** [comment]
What does JHipster do and why would we want it?

**Comment by RickPoleshuck** [comment]
As I discussed last Friday, the Aggregtor needs to be redone. Here are
some of the issues that I see:
1. The Aggregator currently has no authentication or authorization.  2.
The Aggregator is currently Swagger based. You write the definition of
the API as an xml file and it generates Java code. IMO this is great
for prototypes, but is very limited for a full application. It should
be rewritten with Java classes that have Swagger annotations that
generate the Swagger API documentation.3. The only unit test so far is
for the HealthEndpoint4. Functionality is very limited. For example,
the API call to create an attorney will just create a second record if
the same username is used twice. Cloudant/CouchDB makes it difficult to
fix this properly. I believe that Cloudant may be the correct tool for
Open-Sentencing-Model, but for maintaining attorneys and cases that
will be updated frequently, it is the wrong tool.  For open source
projects, I very much like mysql, but I liked IBMs DB2 when I used it
at JPMorgan. (decades ago.) 

jHipster is mostly a tool that creates a starting point for a ReST
service application with many of the industry standards. For example,
without using jHipster I would still want to use the following tools
that jHipster builds with one command line:

1. Spring boot.2. JPA/Hibernate3. Swagger for documentation4.
Checkstyle for static code quality5. Maven6. Tunable logging7. Health
endpoint8. Administration tool9. Spring based authorization of ReST
endpoints.10. Angular front end11. Liquibase to allow SQL Schema
changes to be committed to source control12. Keycloak for oauth2......
But jHipster also has a tool that starts with an entity definition and
with a single command builds, model classes, ReST classes, Angular
front end web pages, both back end and front end unit and integration
tests.
I am planning on a demonstration of a start of rewriting Aggregator
with jHipster on Friday. As I said at last Friday's meeting, IBM is
providing a lot of time and money on this project, and I totally
understand if IBM wants to show off their own tools.
